src/pushsource/type_aliases.pyi

```diff
@@ -0,0 +1,17 @@
+import datetime
+from collections import Mapping, Sequence
+from numbers import Number
+from typing import Union, Collection, Optional
+
+Date = datetime.date
+DateTime = datetime.datetime
+
+MaybeString = Union[str, Collection[str]]
+
+# First attempt at a JsonObject type based on the spec from
```

Comment:
I commented elsewhere as well, but repeating here.  Where this class is used, the underlying APIs are not actually JSON (both ET and koji are being interacted with using xml-rpc).  Although they are pretty close and the type hints might work, it's a bit wrong to be following the JSON spec to come up with the types when there's no JSON in sight.

Elsewhere I said ApiObject, but maybe the top-level type should really be something like Struct or XmlRpcStruct.  I guess what the type is modelling is precisely those types in the "XML-RPC type" table at https://docs.python.org/3/library/xmlrpc.client.html .